src/server/routes/register.ts

```diff
@@ -176,6 +176,18 @@ router.post(
 			const { stateHandle } = encryptedState;
 
 			try {
+				// validate the code contains only numbers and is 6 characters long
+				// The okta api will validate the input fully, but validating here will prevent unnecessary requests
+				if (!/^\d{6}$/.test(code)) {
+					throw new OAuthError(
+						{
+							error: 'api.authn.error.PASSCODE_INVALID',
```

Comment:
Are we happy with the error message text being `'Incorrect code'`? Could we make it a bit more user-friendly, similar to the other `RegistrationError`s?

Reply:
I think that's a future UX improvement that should happen. Most users shouldn't get to this scenario anyway as the client side validation will stop them, this is here as a failsafe, and is the same error that would be returned by okta were an invalid token be supplied.